### ISSUE TITLE
Implement new ML Algorithms

### ISSUE THREAD

**Issue by mayureshagashe2105** [Math-behind-AI/ScratchAI#9]
### Some of the most common ML Algorithms are listed below.  Feel free to suggest other algorithms, not on the list. and we'll update it. 

**Name**
- [ ] Multiple Linear Regression
- [ ] Lasso Regression
- [ ] Ridge Regression
- [ ] Elastic-Net Regression
- [ ] Polynomial Regression
- [ ]  CatBoost Regression
- [ ] Naive Bayes
- [x]  K- nearest neighbors
- [ ] Support vector machines
- [x] K-means
- [ ] K-medoids
- [x] Decision Trees
- [ ] Linear Discriminant Analysis
- [ ] Principal Component Analysis

**Brief  explanation**
- Provide a neat README file in the directory. There are two existing algorithms in the repository, please follow a similar folder structure and document your code with proper markdown syntax/comments.

**Comment by Shreyashmohadikar** [comment]
I would like to contribute by adding the implementation of Naive Bayes
Kindly assign the issue to me 
Thank You

**Comment by HemanthSai7** [comment]
@Shreyashmohadikar Yes, sure! You can continue working on the implementation of the algorithm. Assigning the issue to you.

**Comment by ramyashreeshetty** [comment]
Hello, I would like to contribute. Is it fine if i complete one of the them?

**Comment by HemanthSai7** [comment]
Hi @ramyashreeshetty ,
Sure go ahead. You can select any algorithm which has not already been chosen. Once you are done make a PR.
Thanks.

**Comment by AAnirudh07** [comment]
Hi! I'd like to implement Lasso Regression.

**Comment by mayureshagashe2105** [comment]
Hi @AAnirudh07 ,
Sure go ahead. Once you are done make a PR. Feel free to reach out in case of any queries.
Thanks.

**Comment by kitrak-rev** [comment]
Hi, I would like to implement Linear Discriminant analysis

**Comment by HemanthSai7** [comment]
Sure @kitrak-rev , Assigning this issue to you!🎉Once you are done make a PR. Feel free to reach out in case of any queries.
Thanks.

**Comment by SayanSaha01** [comment]
Hi, I would like to implement Multiple Linear Regression

**Comment by HemanthSai7** [comment]
Hey @SayanSaha01 ,
Sure go ahead. Once you are done make a PR. Feel free to reach out in case of any queries.
Thanks.

**Comment by ArjunwadkarAjay** [comment]
Hi, I would like to implement Principal Component Analysis

**Comment by HemanthSai7** [comment]
Hey @ArjunwadkarAjay ,
Sure go ahead. Once you are done make a PR. Feel free to reach out in case of any queries.
Thanks.

**Comment by TheCodeAlpha26** [comment]
Hi, I would like to contribute by adding k-medoids

**Comment by HemanthSai7** [comment]
Hey @TheCodeAlpha26 ,
Please make a PR once you are done with the implementations.

**Comment by pps-19012** [comment]
Hi, I would like to contribute by adding the implementation of Principal Component Analysis.

**Comment by HemanthSai7** [comment]
Hi @pps-19012 ,
Sure go ahead. Once you are done make a PR. Feel free to reach out in case of any queries.
Thanks.

**Comment by r-o-n-i-t** [comment]
I would like to contribute by adding the implementation of Naive Bayes.
Kindly assign the issue to me.
Thank You

**Comment by mayureshagashe2105** [comment]
Hey @r-o-n-i-t, We cannot assign you to 2 issues simultaneously. Please complete your first assignment and then comment back here. Thank you

**Comment by r-o-n-i-t** [comment]
Ok 
Thank you

**Comment by Shreyashmohadikar** [comment]
> I would like to contribute by adding the implementation of Naive Bayes. Kindly assign the issue to me. Thank You

It's already assigned to me
Kindly check for issues which are not already assigned to anyone

**Comment by r-o-n-i-t** [comment]
@mayureshagashe2105 Please assign me Decision Trees, as I have completed my previous assignment. 
Thank you

**Comment by HemanthSai7** [comment]
Hey @r-o-n-i-t Sure you can continue with the implementation!🔥

**Comment by vandit98** [comment]
hi @HemanthSai7  
I want to add the scratch code for svm and muiltiple linear regression.
Please assign this to me

**Comment by HemanthSai7** [comment]
Hello @vandit98 , Multiple linear regression has already been taken. You can continue with the implementation of SVM. Feel free to reach out if you have any query.
Thanks

**Comment by vandit98** [comment]
hi
i have added the svm scratch code. Also my linear regression code gives a better visualisation and understanding. Kindly have a look.

**Comment by r-o-n-i-t** [comment]
I would like to contribute by adding the implementation of Polynomial Regression.
Kindly assign the issue to me.
Thank You

**Comment by pps-19012** [comment]
Could you please assign Multiple Linear Regression? I have completed my previous task of implementing the PCA and have created a pull request.

**Comment by HemanthSai7** [comment]
Yes sure. assigning you the PR

**Comment by HemanthSai7** [comment]
Hallo contributors,
Since tomorrow is the last day of HacktoberFest, if anyone still wants to create a PR and make it count as a valid PR in Hacktober you are most welcome.🚀Happy open source!
Thanks.

**Comment by radiantherald** [comment]
Hello there, I'm a python programmer well versed in the python functions numpy, pandas and matplotlib. I want to work on the implementation of multiple linear regression. Can I work on it ??

**Comment by HemanthSai7** [comment]
Hello @radiantherald ,
Sure, you can go ahead. Please create a PR once you are done with your implementation. Have a look at the already implemented algorithms for your reference. Also, feel free to reach out if you face any difficulties.
Thanks
